### PR TITLE
Socket connect works with unix file sockets using port null

### PR DESCRIPTION
**Pull request by steinkel**
Socket currently adds the `:port` to the resource, preventing `unix://` file based socket connections.

### CODE REVIEW

**Comment by ADmad** [comment]
If you skip the new method I think this could go to `master` itself.

**Comment by steinkel** [comment]
I'll send it another to master as bugfix and keep it here for the added test in 3.next OK?

**Review by ADmad — APPROVED** [review]

File: src/Network/Socket.php [hunk 1]
```diff
@@ -161,8 +161,13 @@ public function connect()
         }
 
         set_error_handler([$this, '_connectionErrorHandler']);
-        $this->connection = stream_socket_client(
-            $scheme . $this->_config['host'] . ':' . $this->_config['port'],
+        $remoteSocketTarget = $scheme . $this->_config['host'];
+        $port = $this->_config['port'];
+        if (is_int($port)) {
+            $remoteSocketTarget .= ':' . (int)$port;
```
Comment (ADmad) [inline src/Network/Socket.php]: Why the cast to int when already `is_int()` check earlier?  I still preferred the `!== null` check just incase someone sets the port as string.

Comment (steinkel) [reply src/Network/Socket.php]: ok, I'll allow string but cast it to int to ensure we're dealing with an int port

**Comment by markstory** [comment]
Is it ok to close this? Similar changes have been merged into master.

**Comment by steinkel** [comment]
I've extracted the socket instance creation in this branch, and added specific unit tests for the file case.

**Pull request merged**